config/routes.rb

```diff
@@ -0,0 +1,13 @@
+Rails.application.routes.draw do
+  root to: 'homepages#index'
+
+  resources :users, only: [:show, :index]
+  get '/login', to: 'users#login_form', as: 'login'
+  post '/login', to: 'users#login'
+  post '/logout', to: 'users#logout', as: 'logout'
+
+  resources :works
+  post 'works/:id/upvote', to: 'works#upvote', as: 'upvote_work'
```

Comment:
I'm just curious why you didn't do this?

```suggestion
  resources :works do
    resources :votes, only: [:create]
  end
```